---
title: "HIV RNASeq"
author: "Emma Nguyen"
date: "2024-05-05"
output: html_document
---
# Set up working directory
```{r setup}
knitr::opts_knit$set(root.dir = normalizePath("C:/Users/Trang N/Downloads/R Projects files")) 
```

# Install the necessary package and load library
```{r}
library(Seurat)
library(dplyr)
library(readr)
library(DESeq2)
library(ggplot2)
library(EnhancedVolcano)
library(AnnotationDbi)
library("org.Hs.eg.db")
library(ComplexHeatmap)
library(clusterProfiler)
library(gprofiler2) 
library(circlize)
```

# Loading the data
# For bwa alignment
For each file, remove unnecessary columns. Keep only Gene ID(1st column) and read counts (last column) 
```{r}
data_1c <- read.table('counts_1c.txt', header = TRUE)
data_1c <- data_1c[, -c(2:6)]

data_1t <- read.table('counts_1t.txt', header = TRUE)
data_1t <- data_1t[, -c(2:6)]

data_2c <- read.table('counts_2c.txt', header = TRUE)
data_2c <- data_2c[, -c(2:6)]

data_2t <- read.table('counts_2t.txt', header = TRUE)
data_2t <- data_2t[, -c(2:6)]

data_3c <- read.table('counts_3c.txt', header = TRUE)
data_3c <- data_3c[, -c(2:6)]

data_3t <- read.table('counts_3t.txt', header = TRUE)
data_3t <- data_3t[, -c(2:6)]

# Combine all tables into one and remove duplicate column
Data.in <- data.frame(data_1c,data_1t,data_2c,data_2t,data_3c, data_3t)
Data.in <- Data.in[, -c(3,5,7,9,11)]
```

# For Hisat2
Remove unnecessary columns
```{r}
hisat_data <-read.table('hisat_counts.txt', header = TRUE)
hisat_data <- hisat_data[, -c(2:6)]
```
 
## Differential Expression Analysis
# For bwa
```{r}
metadata <- read.csv('HIV_metadata.csv', row.names = 1, header = TRUE)
row.names(Data.in) <- Data.in$Geneid
Data.in <-  Data.in[, -c(1)]

# Rename columns to match metadata 
colnames(Data.in) <- c("SRR24035594","SRR24035593","SRR24035596","SRR24035595","SRR24035598","SRR24035597")

# Re-order columns to match metadata
Data.in <- Data.in[, c("SRR24035593", "SRR24035594", "SRR24035595", "SRR24035596", "SRR24035597", "SRR24035598")]

# Creat DESeq object 
ds_bwa <- DESeqDataSetFromMatrix(countData = Data.in, 
                             colData = metadata,
                             design = ~ Treatment)  

counts(ds_bwa)
```

```{r}
# Only keep data with total per row is >60
subset <-  ds_bwa[(rowSums(counts(ds_bwa)) >= 60),]

# Fix the base to Baseline/ control
relevel(ds_bwa$Treatment, ref='Baseline')

# Run deseq2
deseq_bwa <- DESeq(subset)
res_bwa <- results(deseq_bwa, alpha = 0.05)
head(results(deseq_bwa, tidy=TRUE)) 

# Summary of differential gene expression results
summary(res_bwa)

# Sort summary list by p-value and show the top results
res_bwa <- res_bwa[order(res_bwa$padj),]
head(res_bwa)
```

Volcano Plot
```{r}
par(mfrow=c(1,1))
with(res_bwa, plot(log2FoldChange, -log10(pvalue), pch=20, main="Volcano plot", xlim=c(-3,3)))
with(subset(res_bwa, padj < 0.05), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
with(subset(res_bwa, padj < 0.05 & log2FoldChange > 0), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))
```

PCA of the samples
```{r}
vsdata_bwa <- vst(ds_bwa, blind=FALSE)
plotPCA(vsdata_bwa, intgroup="Treatment") 
```

Convert GeneID to gene names 
```{r}
bwa.df <-  as.data.frame(res_bwa)
bwa.df$symbol <- mapIds(org.Hs.eg.db, keys= rownames(bwa.df), keytype = 'ENSEMBL', colum = "SYMBOL")
bwa.df
```

Make an Enhanced Volcano plot
```{r}
EnhancedVolcano(bwa.df, x='log2FoldChange', y = 'padj', lab = bwa.df$symbol)
summary(bwa.df$adj.P.Val)
bwa.df$padj <- p.adjust(bwa.df$pvalue, method = "BH")
```

# For Hisat2
```{r}
metadata <- read.csv('HIV_metadata.csv', row.names = 1, header = TRUE)
row.names(hisat_data) <- hisat_data$Geneid
hisat_data <- hisat_data[, -c(1)]
colnames(hisat_data) <- c("SRR24035593","SRR24035594","SRR24035595","SRR24035596","SRR24035597","SRR24035598")

# Creat DESeq object 
ds <- DESeqDataSetFromMatrix(countData = hisat_data, 
                             colData = metadata,
                             design = ~ Treatment)
counts(ds)
```

Repeat same treatment as bwa
```{r}
subset <-  ds[(rowSums(counts(ds)) >= 60),]

relevel(ds$Treatment, ref='Baseline')

### Run deseq2
deseq_ob <- DESeq(subset)

res <- results(deseq_ob, alpha = 0.05)
head(results(deseq_ob, tidy=TRUE)) 

summary(res)

##Sort summary list by p-value
res <- res[order(res$padj),]
head(res)
```

Volcano Plot
```{r}
par(mfrow=c(1,1))
with(res, plot(log2FoldChange, -log10(pvalue), pch=20, main="Volcano plot", xlim=c(-3,3)))

with(subset(res, padj < 0.05), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
with(subset(res, padj < 0.05 & log2FoldChange > 0), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))
```

PCA of the samples
```{r}
vsdata <- vst(deseq_ob, blind=FALSE)
plotPCA(vsdata, intgroup="Treatment")
```

Convert GeneID to gene names 
```{r}
sigs.df <-  as.data.frame(res)
sigs.df$symbol <- mapIds(org.Hs.eg.db, keys= rownames(sigs.df), keytype = 'ENSEMBL', colum = "SYMBOL")
sigs.df
```

Make an Enhanced Volcano plot
```{r}
EnhancedVolcano(sigs.df, x='log2FoldChange', y = 'padj', lab = sigs.df$symbol)
summary(sigs.df$adj.P.Val)
sigs.df$padj <- p.adjust(sigs.df$pvalue, method = "BH")
```

PlotCounts to compare the normalized counts between treated and control groups for the top 6 genes
```{r}
par(mfrow=c(2,3))

plotCounts(deseq_ob, gene="ENSG00000213700", intgroup="Treatment")
plotCounts(deseq_ob, gene="ENSG00000111537", intgroup="Treatment")
plotCounts(deseq_ob, gene="ENSG00000205871", intgroup="Treatment")
plotCounts(deseq_ob, gene="ENSG00000243829", intgroup="Treatment")
plotCounts(deseq_ob, gene="ENSG00000214748", intgroup="Treatment")
plotCounts(deseq_ob, gene="ENSG00000124882", intgroup="Treatment")
```

# Base on the Enhanced Volcano plot of bwa vs hisat, select hisat for further downstream analysis
# Generate Heat Map
```{r}
diff.df <- sigs.df 
diff.df <- diff.df %>%
  filter(pvalue < 0.05)

deseq_ob <- estimateSizeFactors(deseq_ob)
mat <- counts(deseq_ob, normalized = TRUE)[rownames(diff.df), ]
mat.z <- t(apply(mat,1, scale))

fig.height = 25
colnames(mat.z) <- colnames(mat)

Heatmap(mat.z, cluster_rows= T, cluster_columns= T, name = "Z-score", row_labels = diff.df[rownames(mat.z),]$symbol, column_names_gp = gpar(fontsize=5.5), row_names_gp = gpar(fontsize=5.5), row_names_rot = 0, column_names_rot = 0)
```

From the heatmap above, further filter to narrow down genes of interest
```{r}
dif2.df <- as.data.frame(diff.df)
dif2.df <- dif2.df %>%
  filter(pvalue < 0.05, baseMean > 100, symbol != "NA")

mat2 <- counts(deseq_ob, normalized = TRUE)[rownames(dif2.df), ]
mat2.z <- t(apply(mat2,1, scale))
colnames(mat2.z) <- colnames(mat2)

Heatmap(mat2.z, cluster_rows= T, cluster_columns= T, name = "Z-score", row_labels = dif2.df[rownames(mat2.z),]$symbol, column_names_gp = gpar(fontsize=10), row_names_gp = gpar(fontsize=10), row_names_rot = 0, column_names_rot = 40)
```

GO Enrichment Analysis 
```{r}
genes_to_test <- rownames(dif2.df)
GO_results <- enrichGO(gene = genes_to_test, OrgDb = "org.Hs.eg.db", keyType = 'ENSEMBL', ont = "BP")
fit <- plot(barplot(GO_results, showCategory=10))
```

Functional Profiling
```{r}
go_upreg <- gost(genes_to_test, organism = 'hsapiens', significant = T)
gostplot(go_upreg, interactive = T)
```

Generate Circos Plot
```{r}
# Generate a .txt file with the data   
write.table(Data.in, file = "bcount.txt", sep = "\t", row.names = FALSE)
count_table <- read.table("bcount.txt", header = TRUE)

# Calculate correlations between genes
cor_matrix <- cor(count_table)

# Create circos plot
circos.initialize(factors = rownames(cor_matrix), xlim = c(0, 1))
chordDiagram(cor_matrix, col = "blue")

# Pick a threshold
threshold <- 0.5
cor_matrix_thresholded <- cor_matrix
cor_matrix_thresholded[abs(cor_matrix) < threshold] <- 0
chordDiagram(cor_matrix_thresholded)

# Create a named vector with group labels for each gene
group_labels <- sample(1:3, length(rownames(cor_matrix)), replace = TRUE)
names(group_labels) <- rownames(cor_matrix)

chordDiagram(cor_matrix, group = group_labels)
```